js/lib/util.js

```diff
@@ -0,0 +1,70 @@
+const asn1 = require('asn1.js');
+
+const Asn1OID = asn1.define('Asn1OID', function () { this.objid(); });
+
+const Asn1PublicKey = asn1.define('PubKeyASN', function () {
+  this.seq().obj(
+    this.key('key').seq().obj(
+      this.key('type').objid(),
+      this.key('curve').objid()
+    ),
+    this.key('point').bitstr()
+  );
+});
+
+
+// ASN1-encoded objects
+exports.OIDNamedCurveP224 = Asn1OID.encode([1, 3, 132, 0, 33]);
+exports.OIDNamedCurveP256 = Asn1OID.encode([1, 2, 840, 10045, 3, 1, 7]);
+exports.OIDNamedCurveP384 = Asn1OID.encode([1, 3, 132, 0, 34]);
+exports.OIDNamedCurveP521 = Asn1OID.encode([1, 3, 132, 0, 35]);
+exports.OIDECPublicKey = Asn1OID.encode([1, 2, 840, 10045, 2, 1]);
+exports.OIDECPrivateKey = Asn1OID.encode([1, 2, 840, 113549, 1, 1, 1]);
+
+
+class AttributeMap {
```

Comment:
We unfortunately can't do anything with pretty printing right now, without maintaining quite a bit of extra code for it, because grpc only accepts objects and objects only allow strings for their keys. So we lose our `Attribute` object when we set an attribute key as the object key since it will be converted into a string at that point. I have setup the `AttributeMap` class so that we can inherit a `Map` in the future, should the grpc module allow it. We can pretty print with a `Map` because `Map` types allow object keys.

Reply:
Quick example of object vs `Map`. This is a `keyTemplate` used for generating a key.

Object output we have now:
```
AttributeMap {
  '1': <Buffer 01>,
  '260': <Buffer 01>,
  '261': <Buffer 01>,
  '262': <Buffer 00>,
  '263': <Buffer 00>,
  '353': <Buffer 00 00 00 00 00 00 00 10>,
  '354': <Buffer 00> }
```

Potential `Map` output:
```
AttributeMap {
  CKA_VALUE_LEN => <Buffer 00 00 00 00 00 00 00 10>,
  CKA_WRAP => <Buffer 00>,
  CKA_UNWRAP => <Buffer 00>,
  CKA_ENCRYPT => <Buffer 01>,
  CKA_DECRYPT => <Buffer 01>,
  CKA_EXTRACTABLE => <Buffer 00>,
  CKA_TOKEN => <Buffer 01> }
```

Just takes a 3 line change to switch it to a `Map`:
```js
// Extend Map
class AttributeMap extends Map {
  constructor(...attributes) {
    // Call Map constructor 
    super();

    attributes.forEach(attr => {
      if (!(attr instanceof Attribute)) {
        throw new Error('Attribute must be an Attribute');
      }

      // Set key and value on Map instead of object
      this.set(attr.key, attr.value);
    });
  }
}
```

No custom inspection needed.